### PR TITLE
version: Replace bespoke application version with new version system

### PR DESCRIPTION
**Pull request by nordicjm**
Uses Zephyr's new version infrastruction system for configuring the version of the application and displaying it.

### CODE REVIEW

**Review by henrikbrixandersen — APPROVED** [review]
🥳

File: app/VERSION [hunk 1]
```diff
@@ -0,0 +1,5 @@
+VERSION_MAJOR = 3
+VERSION_MINOR = 3
+PATCHLEVEL = 99
+VERSION_TWEAK = 0
+EXTRAVERSION =
```
Comment (stephanosio) [inline app/VERSION:5]: The existing implementation used the project version from CMake:

https://github.com/zephyrproject-rtos/example-application/blob/be3bfb0ec3ced11311e32e3c507cd83527b4f3d1/app/CMakeLists.txt#L10

Now, there exists no relation between the reported version and the CMake project version.

If we are dropping the use of CMake project version, we might as well drop the `VERSION 1.0.0` from `app/CMakeLists.txt`.

**Review by stephanosio — DISMISSED** [review]

File: app/VERSION (hunk 1 above)
Comment (nordicjm) [reply app/VERSION]: Changed it to have the same version of 1.0.0

File: app/CMakeLists.txt [hunk 2]
```diff
@@ -7,9 +7,6 @@
 cmake_minimum_required(VERSION 3.13.1)
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 
-project(app LANGUAGES C VERSION 1.0.0)
```
Comment (gmarull) [inline app/CMakeLists.txt:10]: One comment: until now, one could have a module with N apps, each with their own version (using CMake). Maybe not relevant in practice, but worth mentioning.

Comment (nordicjm) [reply app/CMakeLists.txt:10]: This was brought up in https://github.com/zephyrproject-rtos/zephyr/issues/56605 as a limitation of this method as opposed to the other method. Another limitation of this method as opposed to the other offered implementation is that you cannot override versions of applications unless you change the file in the application tree, so if you add a module to your manifest and include that module to someone else's application, you are unable to change the version that the built application shows as.

**Review by carlescufi — APPROVED** [review]
Nice, thanks @nordicjm

**Pull request merged**